pkg/controllers/management/auth/roletemplates/roletemplate_handler.go

```diff
@@ -91,6 +74,66 @@ func (r *roleTemplateHandler) OnChange(_ string, rt *v3.RoleTemplate) (*v3.RoleT
 	return rt, nil
 }
 
+// getClusterRoles gets all the management plane cluster roles needed for this RoleTemplate.
+func (r *roleTemplateHandler) getClusterRoles(rt *v3.RoleTemplate, rules []rbacv1.PolicyRule) ([]*rbacv1.ClusterRole, error) {
+
+	var clusterRoles []*rbacv1.ClusterRole
+	if rt.Context == "cluster" {
```

Comment:
I would make this procedure more "explicit".
Instead of checking `if rt.Context == "cluster"`, I would use a switch case instead.
Eg.
```
switch rt.Context {
case "cluster":
    ...
case "project":
    ...
}
return clusterRoles, nil
```
Wdyt?